### PR TITLE
Create new ride_id_t typedef

### PR DESCRIPTION
**Pull request by IntelOrca**
In order to increase the maximum number of rides, we need to ensure we can change everywhere it is stored as a `uint8_t` into something bigger.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
This perhaps also a nice opportunity to ensure that MAX_RIDES and RIDE_ID_NULL are properly used everywhere.

**Comment by ZehMatt** [comment]
Another nitpick would be to get rid of NetworkRideId_t in NetworkTypes.h as its just used for serialisation purposes, DataSerialiserTraits can use ride_id_t instead.

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 1]
```diff
@@ -646,7 +646,7 @@ static void peep_pathfind_heuristic_search(
         if (tileElement->flags & TILE_ELEMENT_FLAG_GHOST)
             continue;
 
-        uint8_t rideIndex = 0xFF;
+        ride_id_t rideIndex = 0xFF;
```
Comment (Gymnasiast) [inline src/openrct2/peep/GuestPathfinding.cpp]:
```suggestion
        ride_id_t rideIndex = RIDE_ID_NULL;
```

File: src/openrct2/ride/TrackDesign.cpp [hunk 2]
```diff
@@ -1826,7 +1826,7 @@ static bool track_design_place_preview(rct_track_td6* td6, money32* cost, uint8_
     }
 }
 
-static money32 place_track_design(int16_t x, int16_t y, int16_t z, uint8_t flags, uint8_t* outRideIndex)
+static money32 place_track_design(int16_t x, int16_t y, int16_t z, uint8_t flags, ride_id_t* outRideIndex)
 {
     *outRideIndex = 255;
```
Comment (Gymnasiast) [inline src/openrct2/ride/TrackDesign.cpp]:
```suggestion
    *outRideIndex = RIDE_ID_NULL;
```

File: src/openrct2/world/Footpath.cpp [hunk 3]
```diff
@@ -1581,13 +1581,13 @@ void footpath_connect_edges(int32_t x, int32_t y, TileElement* tileElement, int3
 
     if (tileElement->GetType() == TILE_ELEMENT_TYPE_PATH && tileElement->AsPath()->IsQueue())
     {
-        int32_t rideIndex = -1;
+        ride_id_t rideIndex = RIDE_ID_NULL;
         uint8_t entranceIndex = 255;
         for (size_t i = 0; i < neighbourList.count; i++)
         {
             if (neighbourList.items[i].ride_index != 255)
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp]:
```suggestion
            if (neighbourList.items[i].ride_index != RIDE_ID_NULL)
```

File: src/openrct2/world/Footpath.cpp [hunk 4]
```diff
@@ -1744,7 +1744,7 @@ void footpath_queue_chain_reset()
  *
  *  rct2: 0x006A76E9
  */
-void footpath_queue_chain_push(uint8_t rideIndex)
+void footpath_queue_chain_push(ride_id_t rideIndex)
 {
     if (rideIndex != 255)
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp]:
```suggestion
    if (rideIndex != RIDE_ID_NULL)
```

**Comment by ZehMatt** [comment]
Any reason you left out the network types?

**Comment by ZehMatt** [comment]
Oh well, the idea was to simply replace NetworkRideId_t with your new type on all instances, that should have done the job, but I can do it later this day maybe, no worries.

**Pull request merged**